controllers/workspace/devworkspace_controller.go

```diff
@@ -59,6 +59,10 @@ import (
 	dw "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
 )
 
+const (
+	startingWorkspaceRequeueInterval = 5 * time.Second
```

Comment:
I wonder if 5 secs is always the best choice or we can make it dynamic and like to minute, 2,3 in case of 5minutes timeout?

Reply:
I mostly picked 5 seconds because it feels long enough that it's not burdening the controller (we can reconcile many times a second) but also short enough to avoid strange issues in setting a timeout (e.g. if we set a 1 minute requeue, what happens if the config specifies a timeout of 1 minute 15 seconds?)

Reply:
> 1 minute 15 seconds

I don't think there is a case for such a precise timeout. Maybe we can declare our precision to like 1minute or 30 seconds.

And actually, I thought about reconciling after (timeout + last transition time - now + 5 sec), which will initiate reconcile loop after 5 sec when potentially we need to kill that.

But I'm OK with any if it does not generate redundant loading.

Reply:
I tested this informally by starting a workspace that will timeout (ignore `FailedScheduling` events and ask for 100Gi in Theia). While the workspace is looping on checking the deployment every 5 seconds, I ran
```
for i in {1..5}; do
  kubectl patch dw theia-next --type merge -p '{"metadata": {"labels": {"touch": "false"}}}'
  sleep 0.5
  kubectl patch dw theia-next --type merge -p '{"metadata": {"labels": {"touch": "true"}}}'
  sleep 0.5
done
```
to trigger 10 reconciles to the object, with the assumption that each of those would _also_ start a `RequeueAfter`. However, after the 10 quick reconciles are completed, the controller goes back to queuing reconciles every five seconds, rather than 10 reconciles every 5 seconds, so it seems like RequeueAfter is cancelled if an event triggers a reconcile.

I agree that 5 seconds may be the wrong value here; we should tweak this later.